doc/mgmt/SONiC_overlay_ConfigDB.md

```diff
@@ -0,0 +1,234 @@
+| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| 1. All existed APIs change to return default value.<br>2. Add new API to get 'real' data from config DB, which not have default value.                                                                 | Less code change, all app will get default value automatically.                                                                                                                        | 1. There are hardcoded default value in many different place, the default value of those code may different with default value from Yang model, so new default value from config DB may cause code bug, this is a potential risk.<br/>2. 3 MB memory per-process because need load Yang model.<br/>3. 0.05 second to load yang model |
+| 1. Write API change: when write data to config DB, also write default value to 'Default_value_DB'.<br/>2. Read API change: read default value from 'Default_value_DB' and merge with config DB result. | 1. Less memory consumption and better performance when only call read API: read API no need to load yang model.<br/>2. Less code change, all app will get default value automatically. | Hardcoded default value code still need cleanup.                                                                                                                                                                                                                                                                                     |
+
+# 3 Error handling
+
+- Load yang model: throw exception when found yang model data issue.
+- swss-common API: if not found Yang model schema data for a given table name, write warning message to syslog.
+
+# 4 Serviceability and Debug
+
+- Debug version will write debug log to syslog.
+
+# 5 Unit Test
+
+- All new code will 100% covered by gtest test case.
+- Add E2E test case for all new APIs.
```

Comment:
>E2E [](http://example.com/codeflow?start=6&length=3)

Is it possible to unit test all the testcase?

Reply:
Yes, I will remove this part.